src/components/tooltip/tooltip.js

```diff
@@ -128,11 +128,22 @@ function MdTooltipDirective($timeout, $window, $$rAF, $document, $mdUtil, $mdThe
         scope.visibleWatcher = scope.$watch('visible', onVisibleChanged );
         scope.$watch('direction', updatePosition );
       }
+
+      // Updates the aria-label when the element text changes. This watch
+      // doesn't need to be set up if the element doesn't have any data
+      // bindings.
+      if (element.text().indexOf($interpolate.startSymbol()) > -1) {
+        scope.$watch(function() {
```

Comment:
A question about the mutation observer: In order to get it to fire for text changes, the `characterData` and `subtree` option need to be enabled. This fires the observer roughly twice as often as before. I think this might end up being slower than the watcher because we'll be calling the function that sets `aria-label` about 5 times on mouseover and about 5 times again on mouseleave.